tests/components/matrix/test_send_message.py

```diff
@@ -1,41 +1,121 @@
 """Test the send_message service."""
 
+import io
+
+import pytest
+
 from homeassistant.components.matrix import (
     ATTR_FORMAT,
     ATTR_IMAGES,
     DOMAIN as MATRIX_DOMAIN,
     MatrixBot,
 )
-from homeassistant.components.matrix.const import FORMAT_HTML, SERVICE_SEND_MESSAGE
+from homeassistant.components.matrix.const import (
+    FORMAT_HTML,
+    FORMAT_NOTICE,
+    FORMAT_TEXT,
+    SERVICE_SEND_MESSAGE,
+)
 from homeassistant.components.notify import ATTR_DATA, ATTR_MESSAGE, ATTR_TARGET
 from homeassistant.core import HomeAssistant
 
 from tests.components.matrix.conftest import TEST_BAD_ROOM, TEST_JOINABLE_ROOMS
 
 
+@pytest.mark.parametrize(
+    ("ids", "data", "expected_attributes"),
+    [
+        (
+            "Text message",
+            {
+                ATTR_MESSAGE: "Test text message",
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_TEXT},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {"msgtype": "m.text", "body": "Test text message"},
+            },
+        ),
+        (
+            "HTML message",
+            {
+                ATTR_MESSAGE: "Test <b>html</b> message",
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_HTML},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {
+                    "msgtype": "m.text",
+                    "body": "Test <b>html</b> message",
+                    "format": "org.matrix.custom.html",
+                    "formatted_body": "Test <b>html</b> message",
+                },
+            },
+        ),
+        (
+            "Bot (notice) message",
+            {
+                ATTR_MESSAGE: "Test bot (notice) message",
+                ATTR_DATA: {ATTR_FORMAT: FORMAT_NOTICE},
+                ATTR_TARGET: list(TEST_JOINABLE_ROOMS.keys()),
+            },
+            {
+                "target_rooms": list(TEST_JOINABLE_ROOMS.keys()),
+                "message_type": "m.room.message",
+                "content": {
+                    "msgtype": "m.notice",
+                    "body": "Test bot (notice) message",
+                },
+            },
+        ),
+    ],
```

Comment:
```suggestion
    ],
    ids=["Text message", "HTML message", "Bot (notice) message"]
```

I meant the following in https://github.com/home-assistant/core/pull/108030#discussion_r1469629632